circuit-codegen-annotations/src/androidMain/kotlin/com/slack/circuit/codegen/annotations/CircuitInject.kt

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2022 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuit.codegen.annotations
+
+import com.slack.circuit.runtime.screen.Screen
+import dagger.hilt.GeneratesRootInput
+import kotlin.reflect.KClass
+
+/**
+ * Android-specific [CircuitInject], applying the Hilt [GeneratesRootInput] annotation.
+ *
+ * For more general information about this annotation, see
+ * [com.slack.circuit.codegen.annotations.CircuitInject]
+ */
+@GeneratesRootInput
```

Comment:
Is this required for hilt to work? I'm not sure I'm comfortable imposing the dependency on anyone using this artifact

Reply:
It is, however as mentioned `compileOnly` should do the trick. If you're not comfortable with this constraint on dependency here, happy to discuss potentially creating a new optional module for hilt-specific annotation. That'll also introduce a nitpick-y improvement in that we can rename the `scope` parameter to `component` to tie in to Dagger vocabulary, so maybe worth doing anyway

Reply:
I think I'm ok with compileOnly. @stagg @kierse any concerns?

Reply:
If it works with `compileOnly` we're all good, do like the idea of keeping all the codegen together given the current overlap.